crates/ra_hir/src/ty/tests.rs

```diff
@@ -268,6 +268,18 @@ fn test(a: A) {
     );
 }
 
+#[test]
+fn infer_tuple() {
+    check_inference(
+        r#"
+fn test() {
+    let a: (u32, &str) = (1, "a");
```

Comment:
the type annotation here kind of makes the type inference for `(1, "a")` unnecessary, except that we can see that `1` gets inferred to `u32`. Can you add another line without a type annotation, maybe even just `("a", a)` or something, to test a bit more explicitly that tuple expressions get the correct type even when not annotated?